.github/workflows/build.yaml

```diff
@@ -0,0 +1,30 @@
+name: Build Docker image
+
+on:
+  push:
+    branches: ['main']
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
+
+    steps:
+    - uses: actions/checkout@v5
```

Comment:
All of these actions need to be hash pinned. See https://llvm.org/docs/CIBestPractices.html.

Reply:
Done in 7ba9006d15e673d72fa9beff84af24e80661d3c8